compiler/passes/convert-uast.cpp

```diff
@@ -978,8 +995,7 @@ struct Converter {
   // Deduced by looking at 'buildForallLoopExpr' calls in for_expr:
   bool isLoopMaybeArrayType(const uast::IndexableLoop* node) {
     return node->isBracketLoop() && !node->index() &&
-        !node->iterand()->isZip() &&
-        !node->stmt(0)->isConditional();
+           !node->iterand()->isZip() ;
```

Comment:
This change seems odd to me. It can't be an array type if it has an `if` in it, right? Why is this included here?

Reply:
This change was due to the test `test/expressions/if-expr/inside-field` failing with the check for conditionals. The offending code is here:
```chapel
record R3 {
  type T;
  var multiple: [1..2] if !isClass(T) then 2*T
                       else [1..2] if isClass(T) then owned T? else T;
}
```

and it results in this error: `error: illegal assignment of type to value`

Maybe there is a better way to handle it?

Reply:
Oh, I see, yes, indeed, that is an array type. IMO it would be good to include this example (or a simplified version of it) as a comment in this code since it was not obvious at first glance that an array type could have an `if` in it at all.